src/test/java/ca/team2706/frc/robot/config/ConfigTest.java

```diff
@@ -1,79 +1,107 @@
      *
-     * @throws IOException               For setting readLine() expectations
-     * @throws NoSuchMethodException     In case the method to get the robot id can't be found
-     * @throws InvocationTargetException In case the method to get the robot id can't be invoked
-     * @throws IllegalAccessException    In case this test was to hacky for Java
-     * @throws NoSuchFieldException      In case the robotId field can't be found
+     * @throws Exception Java reflection exception.
      */
     @Test
-    public void loadFileTest() throws IOException, NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException {
-        new Expectations() {{
-            bufferedReader.readLine();
-            returns("-5", "0", "4", "two");
-        }};
+    public void testGetRobotID() throws Exception {
+        Method getIdMethod = Config.class.getDeclaredMethod("getRobotId");
+        getIdMethod.setAccessible(true);
+
+        resetIdField();
+        writeToRobotIdFile(1);
+        assertEquals(1, (int) getIdMethod.invoke(null));
+
+        resetIdField();
+        writeToRobotIdFile(2);
+        assertEquals(2, (int) getIdMethod.invoke(null));
+
+        resetIdField();
+        writeToRobotIdFile(3);
+        assertEquals(3, (int) getIdMethod.invoke(null));
 
-        Method robotIdMethod = Config.class.getDeclaredMethod("getRobotId");
-        robotIdMethod.setAccessible(true);
+        resetIdField();
+        writeToRobotIdFile(4);
+        assertEquals(4, (int) getIdMethod.invoke(null));
+    }
+
+    /**
+     * Tests to ensure that if the robot ID file cannot be retrieved that the robot's id is zero.
+     */
+    @Test
+    public void testRobotIDIsZeroWhenNotSet() throws IOException, InvocationTargetException, IllegalAccessException, NoSuchMethodException {
+        Files.delete(ID_FILE);
 
-        Field robotIdField = Config.class.getDeclaredField("robotId");
-        robotIdField.setAccessible(true);
+        Method getIdMethod = Config.class.getDeclaredMethod("getRobotId");
+        getIdMethod.setAccessible(true);
 
-        robotIdField.set(null, -1);
-        assertEquals(-5, robotIdMethod.invoke(null));
+        assertEquals("Wrong robot ID found", 0, (int) getIdMethod.invoke(null));
+    }
 
-        robotIdField.set(null, -1);
-        assertEquals(0, robotIdMethod.invoke(null));
+    /**
+     * Sets the given static final variable to a value.
+     *
+     * @param field    The field to be set.
+     * @param newValue The new value for the field.
+     * @throws Exception Thrown when something inevitably goes wrong with reflection.
+     */
+    private static void setFinalStatic(final Field field, final Object newValue) throws Exception {
+        field.setAccessible(true);
 
-        robotIdField.set(null, -1);
-        assertEquals(4, robotIdMethod.invoke(null));
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
```

Comment:
The final modifier was removed in place of the `getRobotId()` method that reads from the robotId variable, or loads the file if it hasn't been set yet.